dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -0,0 +1,90 @@
+class Api::V1::Pd::WorkshopsController < ::ApplicationController
+  load_and_authorize_resource class: 'Pd::Workshop'
+
+  # GET /api/v1/pd/workshops
+  def index
+    if params[:state]
+      @workshops = @workshops.in_state(params[:state])
+    end
+
+    render json: @workshops, each_serializer: Api::V1::Pd::WorkshopSerializer
+  end
+
+  # GET /api/v1/pd/workshops/1
+  def show
+    render json: @workshop, serializer: Api::V1::Pd::WorkshopSerializer
+  end
+
+  # PATCH /api/v1/pd/workshops/1
+  def update
+    adjust_facilitators
+    if @workshop.update(workshop_params)
+      render json: @workshop, serializer: Api::V1::Pd::WorkshopSerializer
+    else
+      render json: {errors: @workshop.errors.full_messages}, status: :bad_request
+    end
+  end
+
+  # POST /api/v1/pd/workshops
+  def create
+    @workshop.organizer = current_user
+    adjust_facilitators
+    if @workshop.save
+      render json: @workshop, serializer: Api::V1::Pd::WorkshopSerializer
+    else
+      render json: {errors: @workshop.errors.full_messages}, status: :bad_request
+    end
+  end
+
+  # DELETE /api/v1/pd/workshops/1
+  def destroy
+    @workshop.destroy!
+    head :no_content
+  end
+
+  # POST /api/v1/pd/workshops/1/start
+  def start
+    section = @workshop.start!
+    render json: {section_id: section.id, section_code: section.code}
+  end
+
+  # POST /api/v1/pd/workshops/1/end
+  def end
+    @workshop.end!
+    head :no_content
+  end
+
+  private
+
+  def adjust_facilitators
+    facilitator_param_list = params[:pd_workshop].delete(:facilitators)
```

Comment:
I'm open to other suggestions. I'm calling each item in the list `facilitator_params`, so the containing array is `facilitator_param_list`. I could call it `facilitators_params` but that seems even more confusing. Other ideas?